syntax = "proto3";
package proto;

//実際にコマンドから出力されるオブジェクトの型
message Output {
  repeated BotMsg msgs = 1;
}

//発言ひとつ分
message BotMsg {
  repeated OutputMedia medias = 1;
  bool embed_recommended = 2;
}

//データ
message OutputMedia{

  //データタイプ
  enum MediaType{
    UNKNOWN = 0;
    EXTEND = 1; //データタイプの決定を先伸ばし, 直後のデータと連結する
    UTF8 = 2; //文字列
    FILE = 3; //ファイル
    FILE_URL = 4; //ファイルのURL
  }
  MediaType type = 1;

  //データ本体
  bytes data = 2;
  //ファイル名(拡張子あり), ファイルタイプは拡張子で判断
  string filename = 3;

  //直後のデータに同フィールドが続くかどうか
  bool extend_field = 4;
  /*見出しレベル
    0は通常, 1最も強調, 2,3,...と準じる
   */
  int32 level = 5;

  //エラーコード, 0ならエラーなし
  uint32 error = 6;
  //隠せるなら隠すかどうか
  bool spoiled = 7;
}
